addons/web/static/tests/views/kanban/kanban_compiler_tests.js

```diff
@@ -0,0 +1,151 @@
+/** @odoo-module **/
+import { KanbanCompiler } from "@web/views/kanban/kanban_compiler";
+
+function compileTemplate(arch) {
+    const parser = new DOMParser();
+    const xml = parser.parseFromString(arch, "text/xml");
+    const compiler = new KanbanCompiler();
+    return compiler.compile(xml.documentElement).outerHTML;
+}
+
+function areEquivalent(assert, template1, template2) {
+    if (template1.replace(/\s/g, "") === template2.replace(/\s/g, "")) {
+        assert.ok(true);
+    } else {
+        assert.strictEqual(template1, template2);
+    }
+}
+
+QUnit.module("Kanban Compiler", () => {
+    QUnit.test("simple test", async (assert) => {
+        const arch = `
+            <kanban class="o_kanban_test">
+                <templates><t t-name="kanban-box">
+                    <div>
+                        <t t-esc="record.foo.value"/>
+                        <field name="foo"/>
+                    </div>
+                </t></templates>
+            </kanban>`;
+        const expected = `
+            <t><kanban class="o_kanban_test">
+                <templates><t>
+                    <div t-att-tabindex="props.record.model.useSampleModel?-1:0" role="article" t-att-data-id="props.canResequence and props.record.id" t-on-click="onGlobalClick" t-att-class="getRecordClasses()">
+                        <t t-esc="record.foo.value"/><span t-esc="record[&quot;foo&quot;].value"/>
+                    </div>
+                </t></templates>
+            </kanban></t>
+        `;
+
+        areEquivalent(assert, compileTemplate(arch), expected);
+    });
+
+    QUnit.test("advanced test", async (assert) => {
```

Comment:
I'm not fond of compiler tests. It's powerful, but they need good tooling otherwise they're just a pain to maintain. For instance, in owl, the tests snapshots are great, because it's a jest feature that allow to automatically take and update snapshots. Here, each time you do a slight modification in the compiler, you have to edit those tests by hand. I would simply to another kanban test for this.

Reply:
I already did a similar fix/test, see "field with widget and dynamic attributes in kanban". Also, should we factorize the code that does the string interpolation?